source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -727,6 +727,31 @@ TEST_F(NiDAQmxDriverApiTests, ChannelWithDoneEventRegistered_RunCompleteFiniteAc
   EXPECT_EQ(DAQmxSuccess, response.status());
 }
 
+TEST_F(NiDAQmxDriverApiTests, ChannelWithDoneEventRegisteredTwice_RunCompleteFiniteAcquisition_DoneEventResponseIsReceived)
+{
+  create_ai_voltage_chan(0.0, 1.0);
+  ::grpc::ClientContext reader_context;
+  auto reader = register_done_event(reader_context);
+  reader->WaitForInitialMetadata();
```

Comment:
An unfortunate side effect of the separate completion queue for async methods is that ordering of the initial call is not guaranteed. So, you can get intermittent "can't register events on task that is already started" errors. `WaitForInitialMetadata` on the client side seems like an effective workaround. I'm not sure if there's a way to address that in the server.

Reply:
Actually, I'm not sure that this is because of the separate completion queue. It could just be caused by using an async client.